x/leverage/keeper/collateral.go

```diff
@@ -78,9 +78,67 @@ func (k Keeper) GetBorrowerCollateral(ctx sdk.Context, borrowerAddr sdk.AccAddre
 	return totalCollateral
 }
 
-// GetEligibleLiquidationTargets returns a list of borrower addresses eligible for liquidation
+// addressExistInList returns true if address exist in the list of addresses.
+func addressExistInList(list []sdk.AccAddress, addrToCheck sdk.AccAddress) bool {
+	for _, addr := range list {
+		if addrToCheck.Equals(addr) {
+			return true
+		}
+	}
+
+	return false
+}
+
+// GetEligibleLiquidationTargets returns a list of borrower addresses eligible for liquidation.
 func (k Keeper) GetEligibleLiquidationTargets(ctx sdk.Context) ([]sdk.AccAddress, error) {
-	// TODO: Complete this to support liquidation tool
 	// ref: https://github.com/umee-network/umee/issues/229
-	return nil, nil
+	store := ctx.KVStore(k.storeKey)
+	borrowPrefix := types.CreateLoanKeyNoAddress()
+
+	iter := sdk.KVStorePrefixIterator(store, borrowPrefix)
+	defer iter.Close()
+
+	addressEligibleToLiquidation := []sdk.AccAddress{}
+
+	// Iterate over all open borrows, accruing interest on each and collecting new
+	// reserves.
+	for ; iter.Valid(); iter.Next() {
+		// key is borrowPrefix | lengthPrefixed(borrowerAddr) | denom | 0x00
+		key, _ := iter.Key(), iter.Value()
+
+		// remove prefix | denom and null-terminator
+		borrowerAddr := types.GetAddressFromKeyWithPrefix(key, borrowPrefix)
+
+		// if the address is already on the list it can move to the next
+		if addressExistInList(addressEligibleToLiquidation, borrowerAddr) {
+			continue
+		}
```

Comment:
Instead of this approach, create `checkedAddrs := map[string]bool{}` before the loop.

Then here you can 
```go
if checkedAddrs[borrowerAddr.String()] {
  continue
}
checkedAddrs[borrowerAddr.String()] = true
```
This removes the need for the auxiliary function, and also prevents us from checking addresses which are _not_ eligible for liquidation multiple times.

Reply:
done at https://github.com/umee-network/umee/pull/385/commits/8ec8049b16262ca103a64a84fe62a6317ce4debd